### PR TITLE
package rename and related fixes

### PR DESCRIPTION
**Pull request by sreev**

### CODE REVIEW

**Comment by nlu90** [comment]
+1 for instance & api's java part.
cpp & python part seem no changes needed

**Comment by erenavsarogullari** [comment]
+1 for `Scala/Java Streamlet API` parts via following verification steps:
- API Files has been checked and related files are under `org.apache.*`
- Local clean build passed
- All UTs passed
- `Examples` are under `org.apache.*`
 Also following examples have been run locally successfully by verifying their behaviours on UI as well
```
// Scala Streamlet
ScalaIntegerProcessingTopology
ScalaWindowedWordCountTopology
ScalaClassicalMusicTopology
ScalaRepartitionTopology
ScalaTransformsAndCloneTopology

// Java Streamlet
FilesystemSinkTopology
FormattedOutputTopology
IntegerProcessingTopology
RepartitionTopology
```

File: eco-heron-examples/src/java/org/apache/heron/examples/eco/EvenAndOddBolt.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+//  Copyright 2018 Twitter. All rights reserved.
```
Comment (nwangtw) [inline eco-heron-examples/src/java/org/apache/heron/examples/eco/EvenAndOddBolt.java]: Quick questions:

"Copyright 2018 Twitter. All rights reserved."

1. Do we need to change this to Apache as well?
2. How about the year number?

Comment (sreev) [reply eco-heron-examples/src/java/org/apache/heron/examples/eco/EvenAndOddBolt.java]: i want to keep the any other changes (including copyright that you suggested) separate from package renaming.
also, i am waiting to hear back from Mentors, if a long single would be convenient or the current way.

**Comment by sreev** [comment]
@erenavsarogullari and @kramasamy , would you please look in to the travis error.

> [2018-04-08 04:38:59 +0000] [INFO]: Using cluster definition in /home/travis/.heron/conf/local
> Error: Could not find or load main class org.apache.heron.integration_test.topology.scala_streamlet_with_filter_and_transform.ScalaStreamletWithFilterAndTransform
> [2018-04-08 04:38:59 +0000] [ERROR]: Failed to create topology definition file when executing class 'org.apache.heron.integration_test.topology.scala_streamlet_with_filter_and_transform.ScalaStreamletWithFilterAndTransform' of file '/home/travis/.herontests/lib/scala-integration-tests.jar'
> [2018-04-08 04:38:59 +0000] [ERROR]: Unable to submit the topology
> [2018-04-08 04:38:59 +0000] [ERROR]: Failed to submit 20180408043859_IntegrationTest_ScalaStreamletWithFilterAndTransform_c0e1aa5e-481f-4645-b7ab-9519fb34f497 topology :: Traceback (most recent call last):
>   File "integration_test/src/python/test_runner/main.py", line 166, in run_test
>     params.release_package_uri, args)
>   File "integration_test/src/python/test_runner/main.py", line 251, in submit_topology
>     raise status.TestFailure("Unable to submit the topology")
> TestFailure: ('Unable to submit the topology', None)
> [2018-04-08 04:38:59 +0000] [ERROR]: FAILURE: 1/1 tests failed:
> [2018-04-08 04:38:59 +0000] [ERROR]:   - [0s]    : 20180408043859_IntegrationTest_ScalaStreamletWithFilterAndTransform_c0e1aa5e-481f-4645-b7ab-9519fb34f497

**Comment by erenavsarogullari** [comment]
@sree, Following changes needs to be `org.apache.*` as well on your branch. Also i tested these changes locally by running integration-tests.

**1- File:** /heron/integration_test/src/scala/BUILD
```
[Row 12] srcs = glob(["com/twitter/heron/integration_test/**/*.scala"]),
[Row 20] main_class = "com.twitter.heron.integration_test.topology.scala_streamlet_with_filter_and_transform.ScalaStreamletWithFilterAndTransform"
```

**2- File:** /scripts/packages/BUILD
```
[Row 352] strip_prefix = '/integration_test/src/scala/com/twitter/heron/integration_test/topology/'
```

**3- File:** scripts/run_integration_test.sh
```
[Row 11] SCALA_TESTS_DIR="integration_test/src/scala/com/twitter/heron/integration_test/topology"
```

**4- File:** integration_test/src/scala/org/apache/heron/integration_test/common/ScalaIntegrationTestBase.scala
```
//Please add the following lines
import org.apache.heron.streamlet.scala.Builder
import org.apache.heron.streamlet.scala.impl.BuilderImpl
```
**5- File:** 
integration_test/src/scala/org/apache/heron/integration_test/topology/scala_streamlet_with_filter_and_transform/ScalaStreamletWithFilterAndTransform.scala
```
//Please remove the following line
import org.apache.heron.integration_test.common.AbstractTestTopology
```
```
//Please add the following lines
import org.apache.heron.integration_test.common.{
  AbstractTestTopology,
  ScalaIntegrationTestBase
}
import org.apache.heron.streamlet.scala.{Builder, SerializableTransformer}
```

**Comment by sreev** [comment]
Thank you for looking into it and for detailed fixes, @erenavsarogullari .
I have incorporated all those changes.

**Comment by erenavsarogullari** [comment]
@sreev You' re welcome. Travis is Green now ;)

**Comment by sreev** [comment]
@ashvina 
dhalion:
1. checkin `metrics` folder in microsoft/dhalion repo.
2. update the version tag from 0.2.1 to 0.2.2.
3. publish in maven repo.

heron:
1. update version tag in file WORKSPACE.
2. test locally.
3. create another pull request and coordinate so it merged into master.

I am blocked until then.

**Comment by nwangtw** [comment]
Cool. Maybe create an issue for those 3 tests?

**Comment by sreev** [comment]
Thank you for merging, @nlu90.

**Pull request merged**